iceoryx_hoofs/include/iceoryx_hoofs/cxx/helplets.hpp

```diff
@@ -335,6 +335,50 @@ constexpr T from(const F value) noexcept;
 template <typename T, typename F>
 constexpr T into(const F value) noexcept;
 
+/// @brief Macro which generates a setter method useful for a builder pattern.
+/// @param[in] type the data type of the parameter
+/// @param[in] name the name of the parameter
+/// @param[in] defaultValue the default value of the parameter
+/// @code
+///   class MyBuilder {
+///     IOX_BUILDER_PARAMETER(TypeA, NameB, ValueC)
+///     // START generates the following code
+///     private:
+///       TypeA m_NameB = ValueC;
+///
+///     public:
+///       decltype(auto) NameB(TypeA const& value) &&
+///       {
+///           m_NameB = value;
+///           return std::move(*this);
+///       }
+///
+///       decltype(auto) NameB(TypeA&& value) &&
+///       {
+///           m_NameB = std::move(value);
+///           return std::move(*this);
+///       }
+///     // END
+///   };
+/// @endcode
+#define IOX_BUILDER_PARAMETER(type, name, defaultValue)                                                                \
+  private:                                                                                                             \
+    type m_##name = defaultValue;                                                                                      \
+                                                                                                                       \
+  public:                                                                                                              \
+    decltype(auto) name(type const& value)&&                                                                           \
+    {                                                                                                                  \
+        m_##name = value;                                                                                              \
+        return std::move(*this);                                                                                       \
+    }                                                                                                                  \
```

Comment:
I saw that you added doxygen documentation to the parameter created with this macro. Wouldn't it therefore be better to have the function at the top? maybe even as
```
public: decltype(auto) name(type const& value)&& ...
```
This should map the documentation to the function.